extensions/msal-node-extensions/src/lock/CrossPlatformLock.ts

```diff
@@ -32,18 +32,17 @@ export class CrossPlatformLock {
     /**
      * Locks cache from read or writes by creating file with same path and name as
      * cache file but with .lockfile extension. If another process has already created
-     * the lockfile, will retry again based on configuration settings set by CrossPlatformLockOptions
+     * the lockfile, will back off and retry based on configuration settings set by CrossPlatformLockOptions
      */
     public async lock(): Promise<void> {
-        const processId = pid.toString();
-        for (let tryCount = 0; tryCount < this.retryNumber; tryCount++)
+        for (let tryCount = 0; tryCount < this.retryNumber; tryCount++) {
             try {
                 this.logger.info(`Pid ${pid} trying to acquire lock`);
                 this.lockFileHandle = await fs.open(this.lockFilePath, "wx+");
 
                 this.logger.info(`Pid ${pid} acquired lock`);
-                await fs.write(this.lockFileHandle, processId);
-                break;
+                await this.lockFileHandle.write(pid.toString());
+                return;
```

Comment:
Nitpick: `return await this.lockFileHandle.write(pid.toString());`

Reply:
fs.write returns the bytes written (in this case, pid.toString()). We don't want to return this from lock.lock(), so that's why return needs to be in it's own line.